src/Agda2Hs/Compile/Name.hs

```diff
@@ -82,7 +98,26 @@ compileQName f
         -> return $ Just $ unArg rt
       _ -> return Nothing
 
-compileModuleName :: Monad m => ModuleName -> m (Hs.ModuleName ())
+    isQualified :: QName -> Hs.ModuleName () -> C IsQualified
+    isQualified f mod = (inverseScopeLookupName f <$> getScope) >>= \case
```

Comment:
Indeed, it was just hiding under a rock.

Reply:
This still doesn't cover the case for `hiding`, which leads us back to the recurrent problem of associating the abstract syntax with the original concrete syntax to retrieve such information. 

I think we can merge this feature as only supporting qualifications, but I want to just keep a note here in case we do find a more complete overall  solution to this problem (e.g. by tweaking Agda to keep such information for needy backends).